definitions/procedures/pulp/remove.rb

```diff
@@ -0,0 +1,99 @@
+module Procedures::Pulp
+  class Remove < ForemanMaintain::Procedure
+    metadata do
+      description 'Remove pulp2'
+      for_feature :pulp2
+
+      confine do
+        check_min_version('katello-common', '4.0')
+      end
+    end
+
+    def sys
+      ForemanMaintain::Utils::SystemHelpers.new
+    end
+
+    def pulp_data_dirs
+      pulp_data_dir = feature(:pulp2).data_dir
+      [
+        pulp_data_dir + '/published',
+        pulp_data_dir + '/content',
+        pulp_data_dir + '/importers',
+        pulp_data_dir + '/uploads',
+        feature(:mongodb).data_dir
+      ]
+    end
+
+    def pulp_packages
+      [
+        'pulp-server', 'python-pulp-streamer', 'pulp-puppet-plugins',
+        'python-pulp-rpm-common', 'python-pulp-common',
+        'pulp-selinux', 'python-pulp-oid_validation',
+        'python-pulp-puppet-common', 'python-pulp-repoauth',
+        'pulp-rpm-plugins', 'python-blinker', 'python-celery',
+        'python-django', 'python-isodate', 'python-ldap',
+        'python-mongoengine', 'python-nectar', 'python-oauth2',
+        'python-pymongo'
+      ]
+    end
+
+    def data_dir_removal_cmds
+      pulp_data_dirs.collect { |dir| "rm -rf #{dir}" }
+    end
+
+    def ask_to_proceed(rm_cmds)
+      question = "\nWARNING: All pulp2 packages will be removed with the following commands:\n" \
+        "\n# yum remove #{pulp_packages.join('  ')}" \
+        "\n# yum remove rh-mongodb34-*" \
+        "\n\nAll pulp2 data will be removed.\n"
+      question += rm_cmds.collect { |cmd| "\n# #{cmd}" }.join
+      question += "\n\nDo you want to proceed?"
+      answer = ask_decision(question, actions_msg: 'y(yes), q(quit)')
+      abort! if answer != :yes
+    end
+
+    def run
+      rm_cmds = data_dir_removal_cmds
+      ask_to_proceed(rm_cmds)
+
+      remove_pulp
+
+      remove_mongo
+
+      delete_pulp_data(rm_cmds)
+    end
+
+    def remove_pulp
+      with_spinner('Stopping pulp2 services') do |spinner|
+        pulp_services = %w[pulp_celerybeat pulp_workers pulp_resource_manager]
+        feature(:service).handle_services(spinner, 'stop', :only => pulp_services)
+      end
+
+      with_spinner('Removing pulp2 packages') do
+        packages_action(:remove, pulp_packages, :assumeyes => true)
+      end
+    end
+
+    def remove_mongo
+      with_spinner('Stopping mongo services') do |spinner|
+        mongo_services = feature(:mongo).services
+        feature(:service).handle_services(spinner, 'stop', :only => mongo_services)
+      end
```

Comment:
Isn't mongo stopped if you remove the RPM package? Don't we also support an external mongo server? Will it still work in that case?

Reply:
Yeah, we can remove all the services commands, good point.   

And yes, there could be an external mongo.  Maybe we should check that directories exist before we suggest/run removing them?